lib_device_control/api/control_shared.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#ifndef __control_shared_h__
+#define __control_shared_h__
+
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * \defgroup control_shared
+ *
+ * The shared API for using the device control library on the device and host
+ * @{
+ */
+
+/**
+ * This is the version of control protocol. Used to check compatibility
+ */
+#define CONTROL_VERSION 0x10
+
+/**
+ * @{
+ * These types are used in control functions to identify the resource id,
+ * command, version, and status.
+ */
+typedef uint8_t control_resid_t;
+typedef uint8_t control_cmd_t;
+typedef uint8_t control_version_t;
+typedef uint8_t control_status_t;
+typedef uint8_t control_ret_t;

```

Comment:
I suspect that the new enum `control_status_t` is used to return a status value for a Write command, hence the change in the comment on line 25.  The fact that the comment lists four things identified and five typedefs exist may lead to some confusion.  If both `control_status_t` and `control_ret_t` are used for status values, it may be worth adding comments on lines 30 and 31 saying how they differ.

Reply:
I will remove that typedef, and I will add a comment in the story about the control status.